ubcpi/ubcpi.py

```diff
@@ -0,0 +1,468 @@
+    )
+
+    rationale_size = Dict(
+        default={'min': 1, 'max': MAX_RATIONALE_SIZE}, scope=Scope.content,
+        help="The minimum and maximum number of characters a student is allowed for their rationale.",
+    )
+
+    correct_answer = Integer(
+        default=1, scope=Scope.content,
+        help="The correct option for the question",
+    )
+
+    correct_rationale = Dict(
+        default={'text': "In the radio series and the first novel, a group of hyper-intelligent pan-dimensional beings demand to learn the Answer to the Ultimate Question of Life, The Universe, and Everything from the supercomputer, Deep Thought, specially built for this purpose. It takes Deep Thought 7.5 million years to compute and check the answer, which turns out to be 42. Deep Thought points out that the answer seems meaningless because the beings who instructed it never actually knew what the Question was." }, scope=Scope.content,
+        help="The feedback for student for the correct answer",
+    )
+
+    stats = Dict(
+        default={'original': {}, 'revised': {}}, scope=Scope.user_state_summary,
+        help="Overall stats for the instructor",
+    )
+    seeds = List(
+        default=[], scope=Scope.content,
+        help="Instructor configured examples to give to students during the revise stage.",
+    )
+
+    sys_selected_answers = Dict(
+        default={}, scope=Scope.user_state_summary,
+        help="System selected answers to give to students during the revise stage.",
+    )
+
+    algo = Dict(
+        default={'name': 'simple', 'num_responses': '#'}, scope=Scope.content,
+        help="The algorithm for selecting which answers to be presented to students",
+    )
+
+    # Declare that we are part of the grading System
+    has_score = True
+
+    ## Everything below is stolen from https://github.com/edx/edx-ora2/blob/master/apps/openassessment/xblock/lms_mixin.py
+    ## It's needed to keep the LMS+Studio happy.
+    ## It should be included as a mixin.
+
+    start = DateTime(
+        default=None, scope=Scope.settings,
+        help="ISO-8601 formatted string representing the start date of this assignment. We ignore this."
+    )
+
+    due = DateTime(
+        default=None, scope=Scope.settings,
+        help="ISO-8601 formatted string representing the due date of this assignment. We ignore this."
+    )
+
+    weight = Float(
+        display_name="Problem Weight",
+        help=("Defines the number of points each problem is worth. "
+              "If the value is not set, the problem is worth the sum of the "
+              "option point values."),
+        values={"min": 0, "step": .1},
+        scope=Scope.settings
+    )
+
+    def has_dynamic_children(self):
+        """Do we dynamically determine our children? No, we don't have any.
+        """
+        return False
+
+    def max_score(self):
+        """The maximum raw score of our problem.
+        """
+        return 1
+
+    def studio_view(self, context=None):
```

Comment:
There is a Studio view mixin that might be better to use. https://github.com/edx/xblock-utils#studioeditablexblockmixin